client/payment-methods-map.tsx

```diff
@@ -31,6 +31,7 @@ export interface PaymentMethodMapEntry {
 	currencies: string[];
 	stripe_key: string;
 	allows_manual_capture: boolean;
+	allows_pay_later: boolean;
```

Comment:
It feels hacky, I'd prefer to omit this property.

Reply:
This suggestion comes from me.
We're following the existing approach similar how `allows_manual_capture` works. This is the source-of-truth of payment method definitions in JS. 

What other option would you prefer? Doing hardcoded `if ( PM in ['affirm', 'afterpay']` ? That's hacky and error prone. 

I'm open to more appropriate naming but the boolean in this map is the right approach.

Reply:
@KarlisJ did guide me with this when I [asked in our team channel](p1685431915076569-slack-C02C00HCJ8J) to find out the best way to reorder. Karlis reached out to me via DM and gave me this hint. 

I found sense in this since this property is reusable for many scenarios where we need to check if a PM is BNPL or otherwise. In my case, I am using this to sort PM's in order of 
* Cards
* BNPL methods
* Other non BNPL APM's ( Bancontact, SEPA, giropay, iDEAL etc. ) 

Happy to adopt any alternative better approach.

Reply:
'allows_manual_capture' is indeed makes sense. No questions there.

@KarlisJ: kinda 'if ( bnpls_enum.values().constains(PM) )', it adds a place where new BNPLs have to be added in the future, though.

I don't have a strong opinion about adopting my suggestion. My goal was to raise a topic for consideration - feel free to take the less hacky one (you are folks closer to the codebase in this context).